samples/switchless/README.md

```diff
@@ -0,0 +1,133 @@
+# Switchless Calls Sample
+
+This sample demonstrates how to make switchless calls to host from inside an enclave. It is built on top of the [`Hello World`](../helloworld/README.md) sample. The addition is a host function `host_helloworld_switchless` which is called from the enclave switchlessly.
+
+It has the following properties:
+
+- Explain the concept of switchless calls
+- Demonstrate how to mark a function as `transition_using_threads` in EDL, and use [`oeedger8r`](https://github.com/openenclave/openenclave/tree/master/docs/GettingStartedDocs/Edger8rGettingStarted.md) tool to compile it
+- Demonstrate how to configure an enclave to enable switchless calls within it
+
+Prerequisite: you may want to read [Common Sample Information](../README.md#common-sample-information) before going further.
+
+## Switchless Calls
+
+In an enclave application, the host makes **ECALL**s into functions exposed by the enclaves it created. Likewise, the enclaves make **OCALL**s into functions exposed by the host that created them. In either case, the execution has to be transitioned from an untrusted environment to a trusted environment, or vice versa. Since the transition is costly due to heavy security checks, it might be more performance advantageous to make the calls **context-switchless**: the caller delegates the function call to a worker thread in the other environment, which does the real job of calling the function and post the result to the caller. Both the calling thread and the worker thread never leave their respective execution contexts during the perceived function call.
+
+The calling thread and the worker thread need to exchange information twice during the call. When the switchless call is initiated, the caller needs to pass the `job` (representing the function call) to the worker thread. And when the call finishes, the worker thread needs to pass the result back to the caller. Both exchanges need to be synchronized.
+
+## How does OE support switchless OCALLs
```

Comment:
Why is there a section called "How does OE support switchless OCALLs" but no equivalent title for ECALLs?

Reply:
It would be great to specifically call out that the current switchless call is only supported for OCALL, that is  "switchless ECALL is not yet implemented"